homeassistant/components/nissan_leaf.py

```diff
@@ -0,0 +1,338 @@
+                else:
+                    _LOGGER.debug("Got location data for Leaf")
+                    self.data[DATA_LOCATION] = location_response
+
+                    _LOGGER.debug("Location Response: ")
+                    _LOGGER.debug(location_response.__dict__)
+            except:
+                _LOGGER.error("Error fetching location info")
+
+        self.signal_components()
+
+    def signal_components(self):
+        dispatcher_send(self.hass, SIGNAL_UPDATE_LEAF)
+
+    def get_battery(self):
+        request = self.leaf.request_update()
+        battery_status = self.leaf.get_status_from_update(request)
+        while battery_status is None:
+            _LOGGER.debug("Battery data not in yet.")
+            time.sleep(5)
+            battery_status = self.leaf.get_status_from_update(request)
+
+        return battery_status
+
+    def get_climate(self):
+        try:
+            request = self.leaf.get_latest_hvac_status()
+            return request
+        except TypeError:
+            return None
+
+    def set_climate(self, toggle):
+        if toggle:
+            request = self.leaf.start_climate_control()
+            climate_result = self.leaf.get_start_climate_control_result(
+                request)
+
+            while climate_result is None:
+                _LOGGER.debug("Climate data not in yet.")
+                time.sleep(5)
+                climate_result = self.leaf.get_start_climate_control_result(
+                    request)
+
+            _LOGGER.debug(climate_result.__dict__)
+
+            self.signal_components()
+
+            return climate_result.is_hvac_running
+        else:
+            request = self.leaf.stop_climate_control()
+            climate_result = self.leaf.get_stop_climate_control_result(request)
+
+            while climate_result is None:
+                _LOGGER.debug("Climate data not in yet.")
+                time.sleep(5)
+                climate_result = self.leaf.get_stop_climate_control_result(
+                    request)
+
+            _LOGGER.debug(climate_result.__dict__)
+
+            self.signal_components()
+
+            return climate_result.is_hvac_running is False
+
+    def get_location(self):
+        request = self.leaf.request_location()
+        location_status = self.leaf.get_status_from_location(request)
+
+        while location_status is None:
+            _LOGGER.debug("Location data not in yet.")
+            time.sleep(5)
+            location_status = self.leaf.get_status_from_location(request)
+
+        return location_status
+
+    def start_charging(self):
+        return self.leaf.start_charging()
+
+
+class LeafEntity(Entity):
+    def __init__(self, car):
+        self.car = car
+
+    @property
+    def device_state_attributes(self):
+        return {
+            'homebridge_serial': self.car.leaf.vin,
+            'homebridge_mfg': 'Nissan',
+            'homebridge_model': 'Leaf'
+        }
+
+    @asyncio.coroutine

```

Comment:
We've migrated to `async`/`await` syntax. Please drop this decorator and prefix async methods with `async` like `async def async_added_to_hass `. Replace any `yield from` with `await`.